lib/rules/camelcase.js

```diff
@@ -57,6 +60,20 @@ module.exports = {
             return name.indexOf("_") > -1 && name !== name.toUpperCase();
         }
 
+        /**
+         * Checks if a parent of a node is an ObjectPattern.
+         * @param {ASTNode} node The node to check.
+         * @returns {boolean} if the node is inside an ObjectPattern
+         * @private
+         */
+        function isInsideObjectPattern(node) {
+            if (!node.parent) {
+                return false;
+            }
+
+            return node.parent.type === "ObjectPattern" || isInsideObjectPattern(node.parent);
```

Comment:
I don't strongly mind, but it would be nice to use iteration instead of recursion here.